snuba/snuba_migrations/replays/0014_urls_index.py

```diff
@@ -0,0 +1,116 @@
+        storage_set=StorageSetKey.REPLAYS,
+        table_name="replays_local",
+        column=Column(
+            "_urls_hashed",
+            Array(
+                UInt(64),
+                Modifiers(materialized="arrayMap(t -> cityHash64(t), urls)"),
+            ),
+        ),
+        after="urls",
+        target=operations.OperationTarget.LOCAL,
+    )
+
+    yield operations.AddColumn(
+        storage_set=StorageSetKey.REPLAYS,
+        table_name="replays_dist",
+        column=Column(
+            "_urls_hashed",
+            Array(
+                UInt(64),
+                Modifiers(materialized="arrayMap(t -> cityHash64(t), urls)"),
+            ),
+        ),
+        after="urls",
+        target=operations.OperationTarget.DISTRIBUTED,
+    )
+
+    yield operations.AddIndex(
+        storage_set=StorageSetKey.REPLAYS,
+        table_name="replays_local",
+        index_name="bf_urls_hashed",
+        index_expression="_urls_hashed",
+        index_type="bloom_filter()",
+        granularity=1,
+        target=operations.OperationTarget.LOCAL,
+    )
+
+    for column in materialized_columns:
+        yield operations.AddColumn(
+            storage_set=StorageSetKey.REPLAYS,
+            table_name="replays_local",
+            column=Column(
+                f"_{column}_hashed",
+                UInt(64, Modifiers(materialized=f"cityHash64({column})")),
+            ),
+            after=column,
+            target=operations.OperationTarget.LOCAL,
+        )
+
+        yield operations.AddColumn(
+            storage_set=StorageSetKey.REPLAYS,
+            table_name="replays_dist",
+            column=Column(
+                f"_{column}_hashed",
+                UInt(64, Modifiers(materialized=f"cityHash64({column})")),
+            ),
+            after=column,
+            target=operations.OperationTarget.DISTRIBUTED,
+        )
+
+        yield operations.AddIndex(
+            storage_set=StorageSetKey.REPLAYS,
+            table_name="replays_local",
+            index_name=f"bf_{column}_hashed",
+            index_expression=f"_{column}_hashed",
+            index_type="bloom_filter()",
+            granularity=1,
+            target=operations.OperationTarget.LOCAL,
+        )
+
+
+def backward_columns_iter() -> Iterator[operations.SqlOperation]:
+    for column in materialized_columns + ["urls"]:
+        yield operations.DropColumn(
+            StorageSetKey.REPLAYS,
+            "replays_local",
+            f"_{column}_hashed",
+            operations.OperationTarget.LOCAL,
+        )
+
+        yield operations.DropColumn(
+            StorageSetKey.REPLAYS,
+            "replays_dist",
+            f"_{column}_hashed",
+            operations.OperationTarget.DISTRIBUTED,
+        )
+
+        yield operations.DropIndex(
+            StorageSetKey.REPLAYS,
+            "replays_local",
+            f"bf_{column}_hashed",
+            target=operations.OperationTarget.LOCAL,
+        )
```

Comment:
I believe you would need to drop the bloom filter index first before dropping the column the index refers to. You can try running the reverse migration locally to verify.

Reply:
Fixed :D